src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.hpp

```diff
@@ -124,6 +128,39 @@ class MGRStrategyBase
       m_ptrLabels[i] = m_labels[i].data();
     }
   }
+
+  /**
+   * @brief Set up BoomerAMG to perform the mechanics F-solve for the first F-relaxation
+   * @param precond the preconditioner wrapper
+   * @param mgrData auxiliary MGR data
+   */
+  void setMechanicsFSolver( HyprePrecWrapper & precond,
```

Comment:
Why did you have to actually move this to the header file?

Reply:
Because now this function in the templated class `MGRStrategyBase`. I think this part will require a refactoring in which `setMechanicsFSolver` could become a free function. `setMechanicsFSolver` will also be renamed and will have counterparts to set the BoomerAMG options for the coarse system in MGR. So, many things are going to change soon and everything will be better organized.